docs/table_design/expression_partitioning.md

```diff
@@ -0,0 +1,256 @@
+
+```SQL
+INSERT OVERWRITE t_recharge_detail1 PARTITION(dt='2022-04-02',city='texas')
+    SELECT * FROM t_recharge_detail2 PARTITION(p20220402_texas);
+```
+
+### View partitions
+
+When you want to view specific information about automatically created partitions, you need to use the `SHOW PARTITIONS FROM <table_name>` statement. The `SHOW CREATE TABLE <table_name>` statement only returns the syntax for expression partitioning that is configured at table creation.
+
+```SQL
+MySQL > SHOW PARTITIONS FROM t_recharge_detail1;
++-------------+-------------------+----------------+---------------------+--------------------+--------+--------------+-----------------------------+-----------------+---------+----------------+---------------+---------------------+--------------------------+----------+------------+----------+
+| PartitionId | PartitionName     | VisibleVersion | VisibleVersionTime  | VisibleVersionHash | State  | PartitionKey | List                        | DistributionKey | Buckets | ReplicationNum | StorageMedium | CooldownTime        | LastConsistencyCheckTime | DataSize | IsInMemory | RowCount |
++-------------+-------------------+----------------+---------------------+--------------------+--------+--------------+-----------------------------+-----------------+---------+----------------+---------------+---------------------+--------------------------+----------+------------+----------+
+| 16890       | p20220401_Houston | 2              | 2023-07-19 17:24:53 | 0                  | NORMAL | dt, city     | (('2022-04-01', 'Houston')) | id              | 6       | 3              | HDD           | 9999-12-31 23:59:59 | NULL                     | 2.5KB    | false      | 1        |
+| 17056       | p20220402_texas   | 2              | 2023-07-19 17:27:42 | 0                  | NORMAL | dt, city     | (('2022-04-02', 'texas'))   | id              | 6       | 3              | HDD           | 9999-12-31 23:59:59 | NULL                     | 2.5KB    | false      | 1        |
++-------------+-------------------+----------------+---------------------+--------------------+--------+--------------+-----------------------------+-----------------+---------+----------------+---------------+---------------------+--------------------------+----------+------------+----------+
+2 rows in set (0.00 sec)
+```
+
+## Limits
+
+- Since v3.1, StarRocks's [shared-data mode](../deployment/deploy_shared_data.md) supports the time function expression and does not support the column expression.
+- Currently, using CTAS to create a table that use the expression partitioning.
+- Currently, using Spark Load to load data to tables that use the expression partitioning is not supported.
+- When `ALTER TABLE <table_name> DROP PARTITION <partition_name>` statement is used to delete a partition created by using the column expression, data in the partition is directly removed and can not be recovered.
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/33d457f1-2177-46b5-a5da-4ee81b2b0e71)
Add "the" following "when" -> the xxx statement